src/sys/socket/mod.rs

```diff
@@ -677,6 +698,52 @@ pub enum ControlMessage<'a> {
     #[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
     ScmCreds,
 
+    /// A message of type `IP_ORIGDSTADDR` which is triggered by setting
+    /// `RecvOrigDstAddrV4` socket option and is used to get original IPv4
+    /// UDP or TCP destination address when actual destination address
+    /// is overwritten by [`TPROXY`][1], which in turn may be used to
+    /// make a transparent proxy server.
+    /// 
+    /// [1]:https://www.kernel.org/doc/Documentation/networking/tproxy.txt
```

Comment:
This is the wrong man page link.  I can't find the correct man page for Linux, but here's the link for FreeBSD: [ip(4)](https://www.freebsd.org/cgi/man.cgi?ip(4)) (note that the section for IP_ORIGDSTADDR has some typos; I just fixed them upstream and in time the link target will be rebuilt).